Add instructions for icon creation on macOS

CleanCut opened the pull request:
This took me over an hour to figure out, so I thought I'd write some documentation so I don't have to figure it out again (because, let's be honest, I'm not going to remember it 😛).

CleanCut commented:
🤔 Hmm, I'm confused. It looks like [the windows icon is being used at build time](https://github.com/NiklasEi/bevy_game_template/blob/600124c9f52a52e502789d992394915b97e92385/build.rs#L8), but then we're using [`assets/textures/bevy.png`](https://github.com/NiklasEi/bevy_game_template/blob/600124c9f52a52e502789d992394915b97e92385/src/main.rs#L32) to set the windows icon _again_ at runtime? Am I understanding correctly?

NiklasEi commented:
I think these do, at least partially, different things. The build script will make Windows display the image as the file icon (and maybe set it in the window, too?).
The run time code sets the tray image on Linux and Windows. I have only been able to check this on Linux. So if you can test on Windows, and verify that we don't need the build time part anymore, we can remove it.

NiklasEi commented on README.md:
```diff
@@ -26,8 +26,13 @@ _Since Bevy is in heavy development, there regularly are unpublished new feature
- 2. Replace `build/macos/icon_1024x1024.png` with a `1024` times `1024` pixel png icon and run `create_icns.sh` (make sure to run the script inside the `macos` directory) - _Warning: sadly this seems to require a mac..._
+ 1. Replace `build/macos/icon_1024x1024.png` with a `1024` times `1024` pixel png icon and run `create_icns.sh` (make sure to run the script inside the `build/macos` directory) - _Warning: sadly this seems to require a mac..._
+ 2. Replace `build/windows/icon.ico` (used for windows executable and as favicon for the web-builds)
+    * On macOS you can create this by:
```
I use GIMP for ico creation, which should work the same on all major platforms. Open the png file and export it as `.ico`.
If that works on macOS, I would prefer a short scentence pointing to GIMP over a platform specific workflow.

CleanCut commented:
I can get access to a Windows machine later today. I'll see what I can find out about the build-time / file / run-time icon behaviors.

CleanCut commented:
Sorry for the delay. Windows is...weird. But I figured it out.

- I confirmed that the top-left-of-the-window icon in Windows 10 is only set by your runtime code.  Unfortunately, it's also an image completely separate from the `build/` directory icons, so it's super easy to accidentally leave it out-of-sync.
- I added a line to `create_icns.sh` to copy a 256x256 version of the icon into `assets/sprite/textures/app_icon.png` and switched the code to use that for the app icon. So now you can change the one icon in `build/macos`, run the script, and _all_ the icons for all the places get generated and used automagically.

CleanCut commented:
(ready for re-review)
